### PR TITLE
#195 IEEE14 with TapChanger Cpp

### PR DESCRIPTION
**Pull request by MarianneSaugierRTE**

### CODE REVIEW

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelTwoWindingsTransformer.cpp [hunk 1]
```diff
@@ -175,8 +175,7 @@ tapChangerIndex_(0) {
     modelPhaseChanger_->setHighStepIndex(phaseTapChanger->getNbTap() - 1. + lowIndex);
     modelPhaseChanger_->setCurrentStepIndex(phaseTapChanger->getCurrentPosition());
   } else if (ratioTapChanger) {
-    modelRatioChanger_.reset(new ModelRatioTapChanger(tfo->getID()));
-    modelRatioChanger_->setV1SupV2(vNom1_ > vNom2_);
+    modelRatioChanger_.reset(new ModelRatioTapChanger(tfo->getID(), tfo->getRatioTapChanger()->getTerminalRefSide()));
```
Comment (adriguir) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelTwoWindingsTransformer.cpp]: As terminalRef is not compulsory into the iidm xsd scheme (minOccurs = 0), what happens if I don't get this attribute? What is returned by the getTerminalRefSide? Shouldn't it be catched into the if/else if loop into the CPP model (see above comment)?

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelRatioTapChanger.cpp [hunk 2]
```diff
@@ -54,31 +54,22 @@ ModelRatioTapChanger::getUpIncreaseTargetU() {
...
     }
-  } else {
+  } else if (side_ == "TWO") {
     if (increaseRate) {
-      upIncreaseTargetU = false;
-    } else {
       upIncreaseTargetU = true;
+    } else {
+      upIncreaseTargetU = false;
     }
```
Comment (adriguir) [inline dynawo/sources/Models/CPP/ModelNetwork/DYNModelRatioTapChanger.cpp]: What if terminalRef not present into iidm file? Shouldn't there be an else that throw an error?

File: nrt/data/IEEE14/IEEE14_WithAutomata/description/IEEE14_WithAutomata.tex [hunk 3]
```diff
@@ -446,6 +447,79 @@ \subsection{Results}
...
+
+The tap-changer is an automaton that changes the tap of a transformer if its monitored terminal voltage goes out of a certain range for a certain amount of time. In this test, the default C++ model is tested.
+
+\subsection{Initial Conditions}
+
+Initial conditions are the same that in the IEEE 14-bus basic test cases.
+
+\subsection{Models}
+
+Transformer 5-6 is equipped with a ratio tap-changer that controls the voltage at its higher terminal (node 5). The role of the tap-changer is to maintain the high terminal voltage within a certain range, independently of what happens on the other terminal. \\
```
Comment (adriguir) [inline nrt/data/IEEE14/IEEE14_WithAutomata/description/IEEE14_WithAutomata.tex]: In the other test cases, the ratio tap-changer is there but it is inactive (regulating = false).

Comment (MarianneSaugierRTE) [reply nrt/data/IEEE14/IEEE14_WithAutomata/description/IEEE14_WithAutomata.tex]: Setting regulating = true is not enough to activate the tap-changer, we have to add the following line:
`<iidm:terminalRef id="_BUS____5-BUS____6-1_PT" side="ONE"/>`

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelRatioTapChanger.cpp (hunk 2 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelRatioTapChanger.cpp]: if terminalRef is not present in the iidm file, no taps are changed

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelTwoWindingsTransformer.cpp (hunk 1 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelTwoWindingsTransformer.cpp]: see #242

File: dynawo/sources/Models/CPP/ModelNetwork/DYNModelRatioTapChanger.cpp (hunk 2 above)
Comment (MarianneSaugierRTE) [reply dynawo/sources/Models/CPP/ModelNetwork/DYNModelRatioTapChanger.cpp]: see #242

**Pull request merged**